client-src/elements/chromedash-form-field.ts

```diff
@@ -1,49 +1,57 @@
-import {LitElement, html, nothing} from 'lit';
+import {SlDetails, SlIconButton, SlInput} from '@shoelace-style/shoelace';
+import {LitElement, TemplateResult, html, nothing} from 'lit';
+import {customElement, property, state} from 'lit/decorators.js';
 import {ref} from 'lit/directives/ref.js';
+
+import {ChromedashApp} from './chromedash-app';
 import './chromedash-textarea';
 import {ALL_FIELDS, resolveFieldForFeature} from './form-field-specs';
 import {getFieldValueFromFeature, showToastMessage} from './utils.js';
+import {Feature, StageDict} from '../js-src/cs-client';
+import {FormattedFeature} from './form-definition';
+
+interface FormFieldValue {
+  name: string;
+  value: any;
+  stageId?: number;
+  feature?: Feature;
+}
 
+@customElement('chromedash-form-field')
 export class ChromedashFormField extends LitElement {
-  static get properties() {
-    return {
-      name: {type: String},
-      index: {type: Number}, // Represents which field this is on the form.
-      stageId: {type: Number},
-      value: {type: String},
-      fieldValues: {type: Array}, // All other field value objects in current form.
-      feature: {attribute: false}, // The rest of the feature being edited.
-      disabled: {type: Boolean},
-      checkboxLabel: {type: String}, // Optional override of default label.
-      shouldFadeIn: {type: Boolean},
-      loading: {type: Boolean},
-      fieldProps: {type: Object},
-      forEnterprise: {type: Boolean},
-      stageType: {type: Number},
-      componentChoices: {type: Object}, // just for the blink component select field
-      checkMessage: {type: String},
-    };
-  }
-
-  constructor() {
-    super();
-    this.name = '';
-    this.index = -1;
-    this.value = '';
-    this.initialValue = '';
-    this.fieldValues = [];
-    /** @type {import('./form-definition').FormattedFeature} */
-    this.feature = {};
-    this.checkboxLabel = '';
-    this.disabled = false;
-    this.shouldFadeIn = false;
-    this.loading = false;
-    this.forEnterprise = false;
-    this.stageId = undefined;
-    this.stageType = undefined;
-    this.componentChoices = {};
-    this.checkMessage = '';
-  }
+  @property({type: String})
+  name = '';
+  @property({type: Number}) // Represents which field this is on the form.
+  index = -1;
+  @property({type: Array}) // All other field value objects in current form.
+  fieldValues: FormFieldValue[] = [];
```

Comment:
It turns out this isn't just an array; lines like https://github.com/GoogleChrome/chromium-dashboard/blob/a92a06d3219c6d5ac4435fd67c4db781f3247f05/client-src/elements/chromedash-guide-metadata-page.js#L205 assign a `.feature` property to it. You can capture that as `FormFieldValue[] & {feature: Feature}`.